rosenpass/src/app_server.rs

```diff
@@ -1288,3 +1303,62 @@ impl crate::api::mio::MioManagerContext for MioManagerFocus<'_> {
         self.0
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn helper_support_ipv4_ipv6(sockets: &Vec<mio::net::UdpSocket>) -> bool {
+        let mut supports_iv4 = false;
+        let mut supports_ipv6 = false;
+        for socket in sockets {
+            if socket.local_addr().unwrap().is_ipv4() {
+                supports_iv4 = true;
+            } else {
+                supports_ipv6 = true;
+                if !socket.only_v6().unwrap() {
+                    supports_iv4 = true;
+                }
+            }
+        }
+        supports_iv4 && supports_ipv6
+    }
+
+    #[test]
+    pub fn test_open_sockets_empty() {
+        let mut sockets = Vec::new();
+        let _ = open_sockets(&mut sockets);
```

Comment:
```suggestion
        assert!(open_sockets(&mut sockets).is_ok());
```

You probably want to test that your function didn't error out early.